### ISSUE TITLE
MA0042: Advises async method to invoke itself to replace call to sync method

### ISSUE THREAD

**Issue by drieseng** [meziantou/Meziantou.Analyzer#613]
### Version of the Meziantou.Analyzer NuGet package

2.0.93

### Rule Identifier

MA0042

### Target Framework

net6.0

### C# Language version

C# 10

### Description

When an async method invokes a sync method, MA0042 checks if there's an async method instead and - when found - reports a diagnostic that advises use of the async method.

When an async method delegates to a sync method, it's likely that the sync method shares the base name of the async method.
In that case, MA0042 actually encourages the user to recursively invoke the async method.

I think a code fragment is better than this explanation :)

### Reproduction Steps

**Minimal code:**

```c#
public static async Task RunAsync()
{
    await Task.Delay(1000).ConfigureAwait(continueOnCapturedContext: false);
    Run();
}

private static void Run()
{
    Console.WriteLine();
}
```

**Expected result:**

MA0042 is not reported.

**Actual result:**

error MA0042: Use 'RunAsync' instead of 'Run' (https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0042.md)

**Comment by meziantou** [comment]
It's a duplicate of https://github.com/meziantou/Meziantou.Analyzer/issues/587. I'm not sure how to correctly handle this case. There are valid use-cases for reporting the diagnostic (recursion, adding an async path in addition to a sync path), but there are also false-positives. Currently, I would advise to disable the rule using a `#pragma warning disable` or an attribute. I'm open to better suggestions!

**Comment by drieseng** [comment]
Only (automatically) suppress the diagnostic in case the proposed async method is the calling method itself?
